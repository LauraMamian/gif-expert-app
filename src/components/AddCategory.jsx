import PropTypes from 'prop-types';

import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ({target}) => {
        setInputValue(target.value);
    };

    const onSubmit = (e) => {
        // console.log('onSubmit');
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;
        
        onNewCategory(inputValue.trim());
        setInputValue('');
    };

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Buscar GIF"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
};