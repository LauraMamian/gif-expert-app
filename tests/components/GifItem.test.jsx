import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GiftItem />', () => {
    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';

    test('Debe hacer match con el Snapshoot ', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })

    test('Debe mostrar la imagen con el URL y el título indicado', () => {
        render(<GifItem title={title} url={url} />);
        // expect (screen.getByRole('img').src).toBe(url);
        // expect (screen.getByRole('img').alt).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe mostrar el título en el componente', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
})
