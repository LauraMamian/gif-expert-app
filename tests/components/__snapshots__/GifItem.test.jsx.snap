// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem /> Debe hacer match con el Snapshoot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un titulo"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
